samples/hello_world/sample.yaml

```diff
@@ -13,6 +13,6 @@ common:
 tests:
   singlethread:
     extra_args: CONF_FILE=prj_single.conf
-    filter: not CONFIG_BT and not CONFIG_GPIO_SCH
+    filter: not CONFIG_BT and not CONFIG_GPIO_SCH and ASSERT == 0
```

Comment:
does this work? shouldn't this be "not CONFIG_ASSERT"?

Reply:
also please file a bug on this and link to the commit message of this patch

Reply:
it does not work as not CONFIG_ASSERT because ASSERT is being added by sanitycheck using -DCONFIG_ASSERT. I want to change this and make sanitycheck add kconfig overlays instead of defining this via cmake

Reply:
Regarding sanity check adding kconfig overlays: Would this be sufficient https://github.com/zephyrproject-rtos/zephyr/issues/5723?